src/codegen/templates/library_interpreter/default_c_function_call.py.mako

```diff
@@ -14,11 +14,22 @@
     samps_per_chan_param = get_samps_per_chan_read_or_write_param(function.base_parameters)
 %>\
         cfunc = lib_importer.${'windll' if function.calling_convention == 'StdCall' else 'cdll'}.DAQmx${function.c_function_name}
+<%
+    argtypes = get_argument_types(function)
+%>\
+%if 'DateTime' in argtypes:
+    <%
+        index = argtypes.index('DateTime')
+        name = function_call_args[index]
+        argtypes[index] = 'ctypes.CVIAbsoluteTime'
+    %>\
+    ${name} = LibTimestamp.from_datetime(${name})
+%endif
```

Comment:
On second thought, generate_interpreter_function_call_args is probably more appropriate: https://github.com/ni/nidaqmx-python/blob/master/src/codegen/utilities/interpreter_helpers.py#L135

get_instantiation_lines_for_output is really for out-parameters.

Reply:
Thanks for the suggestion. 
But since this is an interpreter and the function names being passed are positional arguments, I cannot update function_call_args to do the conversion since that would look like 
`error_code = cfunc(
            task, when=AbsoluteTime.from_datetime(when), timescale)`
and that would result in an error since a positional argument cannot come after a keyword argument.

So I think I need to still have `when = AbsoluteTime.from_datetime(when)` as a separate line in the _library_interpreter file. However I have updated the helpers.py files (generate_interpreter_function_call_args) to do the majority of the parsing jobs and push the name of the dateTime/AbsoluteTime parameters into another list, which will be called and displayed in the mako file.

I have moved the change to argtypes to be directly in to_param_argtype() instead since any special modifications to argument types seem to happen there.

Reply:
> that would result in an error since a positional argument cannot come after a keyword argument.

Right, you need it to use a positional argument: `error_code = cfunc( task, AbsoluteTime.from_datetime(when), timescale)`

> So I think I need to still have when = AbsoluteTime.from_datetime(when) as a separate line in the _library_interpreter file. 

I disagree. In generate_interpreter_function_call_args , you should specify `f"AbsoluteTime.from_datetime({param.parameter_name})"` instead of `f"{param.parameter_name} = AbsoluteTime.from_datetime({param.parameter_name})"` so that you generate a positional argument rather than a keyword argument.

Reply:
Yeah, I think this code is already too complicated and inconsistent, and adding another way to generate variable declarations for parameters makes it worse, not better. Please extend the existing mechanisms.

Reply:
Thanks for clarifying, I understand what you mean now. I have updated the codegen and the file accordingly to just use existing mechanisms. As this PR only covers CfgStartTimeTrigger, I will handle WaitForValidTimestamp in a separate PR.